### PR TITLE
Case 20321: More Entity Property Cleanup

### PR DESCRIPTION
**Pull request by SamGondelman**
https://highfidelity.manuscript.com/f/cases/20321/More-Entity-Property-Cleanup

Just reorganization and removing unused properties for the sake of my sanity and the sanity of others.

Test plan:
- Run the automated protocol tests.

### CODE REVIEW

File: libraries/entities-renderer/src/RenderableLineEntityItem.cpp [hunk 1]
```diff
@@ -37,7 +37,7 @@ void LineEntityRenderer::doRenderUpdateAsynchronousTyped(const TypedEntityPointe
     if (_lineVerticesID == GeometryCache::UNKNOWN_ID) {
         _lineVerticesID = geometryCache->allocateID();
     }
-    glm::vec4 lineColor(toGlm(entity->getColor()), entity->getLocalRenderAlpha());
```
Comment (SamGondelman) [inline libraries/entities-renderer/src/RenderableLineEntityItem.cpp]: This property doesn't work and has been removed.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15965/HighFidelity-Beta-PR14585-ff5cfaa.apk

File: libraries/entities/src/EntityItem.cpp [hunk 2]
```diff
@@ -70,33 +70,60 @@ EntityItem::~EntityItem() {
 EntityPropertyFlags EntityItem::getEntityProperties(EncodeBitstreamParams& params) const {
     EntityPropertyFlags requestedProperties;
 
+    // Core
```
Comment (SamGondelman) [inline libraries/entities/src/EntityItem.cpp]: These changes (and the others) look scary!  But they aren't!  They're just reorganization.  In every place where we do stuff like this, the properties are now in this order:

```
    // Core properties
    PROP_SIMULATION_OWNER,
    PROP_VISIBLE,
    PROP_NAME,
    PROP_LOCKED,
    PROP_USER_DATA,
    PROP_HREF,
    PROP_DESCRIPTION,
    PROP_POSITION,
    PROP_DIMENSIONS,
    PROP_ROTATION,
    PROP_REGISTRATION_POINT,
    PROP_CREATED,
    PROP_LAST_EDITED_BY,
    PROP_ENTITY_HOST_TYPE,
    PROP_OWNING_AVATAR_ID,
    PROP_PARENT_ID,
    PROP_PARENT_JOINT_INDEX,
    PROP_QUERY_AA_CUBE,
    PROP_CAN_CAST_SHADOW,
    PROP_VISIBLE_IN_SECONDARY_CAMERA,
    // Grab
    PROP_GRAB_GRABBABLE,
    PROP_GRAB_KINEMATIC,
    PROP_GRAB_FOLLOWS_CONTROLLER,
    PROP_GRAB_TRIGGERABLE,
    PROP_GRAB_EQUIPPABLE,
    PROP_GRAB_LEFT_EQUIPPABLE_POSITION_OFFSET,
    PROP_GRAB_LEFT_EQUIPPABLE_ROTATION_OFFSET,
    PROP_GRAB_RIGHT_EQUIPPABLE_POSITION_OFFSET,
    PROP_GRAB_RIGHT_EQUIPPABLE_ROTATION_OFFSET,
    PROP_GRAB_EQUIPPABLE_INDICATOR_URL,
    PROP_GRAB_EQUIPPABLE_INDICATOR_SCALE,
    PROP_GRAB_EQUIPPABLE_INDICATOR_OFFSET,

    // Physics
    PROP_DENSITY,
    PROP_VELOCITY,
    PROP_ANGULAR_VELOCITY,
    PROP_GRAVITY,
    PROP_ACCELERATION,
    PROP_DAMPING,
    PROP_ANGULAR_DAMPING,
    PROP_RESTITUTION,
    PROP_FRICTION,
    PROP_LIFETIME,
    PROP_COLLISIONLESS,
    PROP_COLLISION_MASK,
    PROP_DYNAMIC,
    PROP_COLLISION_SOUND_URL,
    PROP_ACTION_DATA,

    // Cloning
    PROP_CLONEABLE,
    PROP_CLONE_LIFETIME,
    PROP_CLONE_LIMIT,
    PROP_CLONE_DYNAMIC,
    PROP_CLONE_AVATAR_ENTITY,
    PROP_CLONE_ORIGIN_ID,

    // Scripts
    PROP_SCRIPT,
    PROP_SCRIPT_TIMESTAMP,
    PROP_SERVER_SCRIPTS,

    // Certifiable Properties
    PROP_ITEM_NAME,
    PROP_ITEM_DESCRIPTION,
    PROP_ITEM_CATEGORIES,
    PROP_ITEM_ARTIST,
    PROP_ITEM_LICENSE,
    PROP_LIMITED_RUN,
    PROP_MARKETPLACE_ID,
    PROP_EDITION_NUMBER,
    PROP_ENTITY_INSTANCE_NUMBER,
    PROP_CERTIFICATE_ID,
    PROP_STATIC_CERTIFICATE_VERSION,

    // Used to convert values to and from scripts
    PROP_LOCAL_POSITION,
    PROP_LOCAL_ROTATION,
    PROP_LOCAL_VELOCITY,
    PROP_LOCAL_ANGULAR_VELOCITY,
    PROP_LOCAL_DIMENSIONS,
    
    // These properties are used by multiple subtypes but aren't in the base EntityItem
    PROP_SHAPE_TYPE,
    PROP_COMPOUND_SHAPE_URL,
    PROP_COLOR,
    PROP_ALPHA,
    PROP_TEXTURES,

    // Particles
    PROP_MAX_PARTICLES,
    PROP_LIFESPAN,
    PROP_EMITTING_PARTICLES,
    PROP_EMIT_RATE,
    PROP_EMIT_SPEED,
    PROP_SPEED_SPREAD,
    PROP_EMIT_ORIENTATION,
    PROP_EMIT_DIMENSIONS,
    PROP_ACCELERATION_SPREAD,
    PROP_POLAR_START,
    PROP_POLAR_FINISH,
    PROP_AZIMUTH_START,
    PROP_AZIMUTH_FINISH,
    PROP_EMIT_RADIUS_START,
    PROP_EMIT_ACCELERATION,
    PROP_PARTICLE_RADIUS,
    PROP_RADIUS_SPREAD,
    PROP_RADIUS_START,
    PROP_RADIUS_FINISH,
    PROP_COLOR_SPREAD,
    PROP_COLOR_START,
    PROP_COLOR_FINISH,
    PROP_ALPHA_SPREAD,
    PROP_ALPHA_START,
    PROP_ALPHA_FINISH,
    PROP_EMITTER_SHOULD_TRAIL,
    PROP_PARTICLE_SPIN,
    PROP_SPIN_START,
    PROP_SPIN_FINISH,
    PROP_SPIN_SPREAD,
    PROP_PARTICLE_ROTATE_WITH_ENTITY,

    // Model
    PROP_MODEL_URL,
    PROP_JOINT_ROTATIONS_SET,
    PROP_JOINT_ROTATIONS,
    PROP_JOINT_TRANSLATIONS_SET,
    PROP_JOINT_TRANSLATIONS,
    PROP_RELAY_PARENT_JOINTS,
    // Animation
    PROP_ANIMATION_URL,
    PROP_ANIMATION_ALLOW_TRANSLATION,
    PROP_ANIMATION_FPS,
    PROP_ANIMATION_FRAME_INDEX,
    PROP_ANIMATION_PLAYING,
    PROP_ANIMATION_LOOP,
    PROP_ANIMATION_FIRST_FRAME,
    PROP_ANIMATION_LAST_FRAME,
    PROP_ANIMATION_HOLD,

    // Light
    PROP_IS_SPOTLIGHT,
    PROP_INTENSITY,
    PROP_EXPONENT,
    PROP_CUTOFF,
    PROP_FALLOFF_RADIUS,

    // Text
    PROP_TEXT,
    PROP_LINE_HEIGHT,
    PROP_TEXT_COLOR,
    PROP_TEXT_ALPHA,
    PROP_BACKGROUND_COLOR,
    PROP_BACKGROUND_ALPHA,
    PROP_BILLBOARD_MODE,
    PROP_LEFT_MARGIN,
    PROP_RIGHT_MARGIN,
    PROP_TOP_MARGIN,
    PROP_BOTTOM_MARGIN,

    // Zone
    // Keylight
    PROP_KEYLIGHT_COLOR,
    PROP_KEYLIGHT_INTENSITY,
    PROP_KEYLIGHT_DIRECTION,
    PROP_KEYLIGHT_CAST_SHADOW,
    // Ambient light
    PROP_AMBIENT_LIGHT_INTENSITY,
    PROP_AMBIENT_LIGHT_URL,
    // Skybox
    PROP_SKYBOX_COLOR,
    PROP_SKYBOX_URL,
    // Haze
    PROP_HAZE_RANGE,
    PROP_HAZE_COLOR,
    PROP_HAZE_GLARE_COLOR,
    PROP_HAZE_ENABLE_GLARE,
    PROP_HAZE_GLARE_ANGLE,
    PROP_HAZE_ALTITUDE_EFFECT,
    PROP_HAZE_CEILING,
    PROP_HAZE_BASE_REF,
    PROP_HAZE_BACKGROUND_BLEND,
    PROP_HAZE_ATTENUATE_KEYLIGHT,
    PROP_HAZE_KEYLIGHT_RANGE,
    PROP_HAZE_KEYLIGHT_ALTITUDE,
    // Bloom
    PROP_BLOOM_INTENSITY,
    PROP_BLOOM_THRESHOLD,
    PROP_BLOOM_SIZE,
    PROP_FLYING_ALLOWED,
    PROP_GHOSTING_ALLOWED,
    PROP_FILTER_URL,
    PROP_KEY_LIGHT_MODE,
    PROP_AMBIENT_LIGHT_MODE,
    PROP_SKYBOX_MODE,
    PROP_HAZE_MODE,
    PROP_BLOOM_MODE,

    // Polyvox
    PROP_VOXEL_VOLUME_SIZE,
    PROP_VOXEL_DATA,
    PROP_VOXEL_SURFACE_STYLE,
    PROP_X_TEXTURE_URL,
    PROP_Y_TEXTURE_URL,
    PROP_Z_TEXTURE_URL,
    PROP_X_N_NEIGHBOR_ID,
    PROP_Y_N_NEIGHBOR_ID,
    PROP_Z_N_NEIGHBOR_ID,
    PROP_X_P_NEIGHBOR_ID,
    PROP_Y_P_NEIGHBOR_ID,
    PROP_Z_P_NEIGHBOR_ID,

    // Web
    PROP_SOURCE_URL,
    PROP_DPI,

    // Polyline
    PROP_LINE_POINTS,
    PROP_STROKE_WIDTHS,
    PROP_STROKE_NORMALS,
    PROP_STROKE_COLORS,
    PROP_IS_UV_MODE_STRETCH,

    // Shape
    PROP_SHAPE,

    // Material
    PROP_MATERIAL_URL,
    PROP_MATERIAL_MAPPING_MODE,
    PROP_MATERIAL_PRIORITY,
    PROP_PARENT_MATERIAL_NAME,
    PROP_MATERIAL_MAPPING_POS,
    PROP_MATERIAL_MAPPING_SCALE,
    PROP_MATERIAL_MAPPING_ROT,
    PROP_MATERIAL_DATA,
    PROP_MATERIAL_REPEAT,

    // Image
    PROP_IMAGE_URL,
    PROP_EMISSIVE,
    PROP_KEEP_ASPECT_RATIO,
    PROP_SUB_IMAGE,

    // Grid
    PROP_GRID_FOLLOW_CAMERA,
    PROP_MAJOR_GRID_EVERY,
    PROP_MINOR_GRID_EVERY,
```

File: libraries/entities/src/EntityItemProperties.cpp [hunk 3]
```diff
@@ -733,16 +786,15 @@ EntityPropertyFlags EntityItemProperties::getChangedProperties() const {
...
  * It has properties in addition to the common {@link Entities.EntityProperties|EntityProperties}.
  * @typedef {object} Entities.EntityProperties-Line
- * @property {Vec3} dimensions=0.1,0.1,0.1 - The dimensions of the entity. Must be sufficient to contain all the 
+ * @property {Vec3} dimensions=0.1,0.1,0.1 - The dimensions of the entity. Must be sufficient to contain all the
  *     <code>linePoints</code>.
  * @property {Vec3[]} linePoints=[]] - The sequence of points to draw lines between. The values are relative to the entity's
- *     position. A maximum of 70 points can be specified. The property's value is set only if all the <code>linePoints</code> 
+ *     position. A maximum of 70 points can be specified. The property's value is set only if all the <code>linePoints</code>
  *     lie within the entity's <code>dimensions</code>.
- * @property {number} lineWidth=2 - <em>Currently not used.</em>
```
Comment (SamGondelman) [inline libraries/entities/src/EntityItemProperties.cpp]: This didn't do anything and we're going to remove Lines soon, so I've removed it from Lines and PolyLines (where it was unused AND unnecessary, because we have PROP_STROKE_WIDTHS).

File: libraries/entities/src/EntityPropertyFlags.cpp [hunk 4]
```diff
@@ -1,208 +0,0 @@
...
-    result =
-        f.getHasProperty(PROP_GRAB_RIGHT_EQUIPPABLE_ROTATION_OFFSET) ? result + "grab.RightEquippableRotationOffset " : result;
-    result = f.getHasProperty(PROP_GRAB_EQUIPPABLE_INDICATOR_URL) ? result + "grab.EquippableIndicatorURL " : result;
-    result = f.getHasProperty(PROP_GRAB_EQUIPPABLE_INDICATOR_SCALE) ? result + "grab.EquippableIndicatorScale " : result;
-    result = f.getHasProperty(PROP_GRAB_EQUIPPABLE_INDICATOR_OFFSET) ? result + "grab.EquippableIndicatorOffset " : result;
-
-    result += "]";
-    dbg.nospace() << result;
-    return dbg;
-}
```
Comment (SamGondelman) [inline libraries/entities/src/EntityPropertyFlags.cpp]: This has been automated and moved to the bottom of EntityItemProperties.cpp

File: libraries/shared/src/AABox.cpp [hunk 5]
```diff
@@ -480,7 +480,7 @@ void AABox::embiggen(const glm::vec3& scale) {
 }
 
 void AABox::setScaleStayCentered(const glm::vec3& scale) {
-    _corner += -0.5f * scale;
+    _corner -= 0.5f * (scale - _scale);
```
Comment (SamGondelman) [inline libraries/shared/src/AABox.cpp]: This was wrong!  It's not used very much so no one noticed.

File: scripts/developer/tests/avatarToWorldTests.js [hunk 6]
```diff
@@ -42,52 +42,6 @@ function jointToWorldPointTest_update(deltaTime) {
    Entities.editEntity(jointToWorldPointTest_sphereEntity, newProperties);
 }
 
-//jointToWorldDirection
```
Comment (SamGondelman) [inline scripts/developer/tests/avatarToWorldTests.js]: These were the only mentions of "Line" entities in our scripts that I could find.  Since they're deprecated, I've removed the tests.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15966/HighFidelity-Beta-PR14585-6094a2f.apk

File: libraries/entities/src/EntityPropertyFlags.h [hunk 7]
```diff
@@ -18,297 +18,303 @@ enum EntityPropertyList {
...
+    PROP_DERIVED_21,
+    PROP_DERIVED_22,
+    PROP_DERIVED_23,
+    PROP_DERIVED_24,
+    PROP_DERIVED_25,
+    PROP_DERIVED_26,
+    PROP_DERIVED_27,
+    PROP_DERIVED_28,
+    PROP_DERIVED_29,
+    PROP_DERIVED_30,
```
Comment (SamGondelman) [inline libraries/entities/src/EntityPropertyFlags.h]: Our old system of adding new derived EntityItem properties was a mess.  Hopefully this will be cleaner.  Now: we maximize the efficiency of adding new sub-class properties by taking advantage of the overlap.

File: libraries/entities/src/EntityPropertyFlags.h [hunk 8]
```diff
@@ -18,297 +18,303 @@ enum EntityPropertyList {
...
+    PROP_RESTITUTION,
+    PROP_FRICTION,
+    PROP_LIFETIME,
     PROP_COLLISIONLESS,
-    PROP_DYNAMIC, // 24
-
-    // property used by Light entity
-    PROP_IS_SPOTLIGHT,
-    PROP_DIFFUSE_COLOR,
-    PROP_AMBIENT_COLOR_UNUSED, // FIXME - No longer used, can remove and bump protocol
```
Comment (SamGondelman) [inline libraries/entities/src/EntityPropertyFlags.h]: if you say so

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15966/HighFidelity-Beta-PR14585-6094a2f.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15966/HighFidelity-Beta-Interface-PR14585-6094a2f.dmg

**Comment by QliCknox** [comment]
Currently Testing

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/16191/HighFidelity-Beta-PR14585-6094a2f.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16191/HighFidelity-Beta-PR14585-6094a2f.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16191/HighFidelity-Beta-Interface-PR14585-6094a2f.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16191/HighFidelity-Beta-PR14585-6094a2f.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16191/HighFidelity-Beta-Interface-PR14585-6094a2f.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/16205/HighFidelity-Beta-PR14585-ea11497.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16205/HighFidelity-Beta-PR14585-ea11497.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16205/HighFidelity-Beta-Interface-PR14585-ea11497.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16205/HighFidelity-Beta-PR14585-ea11497.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16205/HighFidelity-Beta-Interface-PR14585-ea11497.exe

**Comment by qlimguicheteau** [comment]
Now Testing: Pre Merge plan

**Comment by qlimguicheteau** [comment]
Halted Testing
Output requests Server access
[OutputFiles.zip](https://github.com/highfidelity/hifi/files/2724996/OutputFiles.zip)

File: libraries/entities/src/EntityItem.cpp [hunk 9]
```diff
@@ -766,6 +773,14 @@ int EntityItem::readEntityDataFromBuffer(const unsigned char* data, int bytesLef
...
     };
 
+    // Core
+    // PROP_SIMULATION_OWNER handled above
+    READ_ENTITY_PROPERTY(PROP_VISIBLE, bool, setVisible);
+    READ_ENTITY_PROPERTY(PROP_NAME, QString, setName);
+    READ_ENTITY_PROPERTY(PROP_LOCKED, bool, setLocked);
+    READ_ENTITY_PROPERTY(PROP_USER_DATA, QString, setUserData);
+    READ_ENTITY_PROPERTY(PROP_HREF, QString, setHref);
+    READ_ENTITY_PROPERTY(PROP_DESCRIPTION, QString, setDescription);
```
Comment (AndrewMeadows) [inline libraries/entities/src/EntityItem.cpp]: Moving these properties up above position and rotation violates a comment up above that says:

`// pack SimulationOwner, transform, and velocity properties near each other`

Note: that comment is actually in the "unpack" portion of the code when it should have been put in `EntityItem::appendEntityData()`.  In any case, there isn't a hard reason to pack those together.  Instead there was a soft reason: if we every manage to transmit "partial" property updates then a very common subgroup of properties that need to be sent will be **simulationOwner + transforms + velocities** and so it seemed like a good idea to keep those together to make it a little easier to make that transition in the Glorious Future.

That said: we definitely need **simulationOwner** to be packed before **transforms + velocities** so we can decide whether we want to actually keep the **transforms + velocities** data or discard it, depending on the various race conditions of simulation ownership bid state.  I merely raise the point so we continue to be aware.  This PR does not break that.

So, in keeping with the spirit of this PR you should delete that comment about packing those properties next to each other since we are no longer doing it.  I'm fine with removing that comment in a near-future PR, it doesn't need to be this one.

File: libraries/entities/src/EntityItem.cpp [hunk 10]
```diff
@@ -1324,31 +1353,13 @@ EntityItemProperties EntityItem::getProperties(const EntityPropertyFlags& desire
...
-    COPY_ENTITY_PROPERTY_TO_PROPERTIES(cloneAvatarEntity, getCloneAvatarEntity);
-    COPY_ENTITY_PROPERTY_TO_PROPERTIES(cloneOriginID, getCloneOriginID);
-
-    withReadLock([&] {
-        _grabProperties.getProperties(properties);
-    });
+    // FIXME: are these needed?
+    //COPY_ENTITY_PROPERTY_TO_PROPERTIES(localVelocity, getLocalVelocity);
+    //COPY_ENTITY_PROPERTY_TO_PROPERTIES(localAngularVelocity, getLocalAngularVelocity);
+    //COPY_ENTITY_PROPERTY_TO_PROPERTIES(localDimensions, getLocalDimensions);
```
Comment (AndrewMeadows) [inline libraries/entities/src/EntityItem.cpp]: Alas, I don't have a clear picture of where we currently need to distinguish "local" from "world".  Most things don't have a parent and for that case "local" is "world".  Ideally we would only ever send **hierarchy + localTransform** data and the receiving end would compute world-frame transforms only when needed (e.g. in the Glorious Future).  Unfortunately the problem is: the "local" stuff was added after we had already laid some foundation that requires world-frame data in some contexts and we are now in an intermediate stage, which is why this is so confusing.

In short: those local properties are not currently sent, and probably never will be, however this commented out code with **FIXME** question should remain because it is still a confusing point. We'll have to pay this technical debt later.

Perhaps one way to make progress in this PR is to add a long comment here about the fundamental problem rather than to ask the "are these needed?" question.

File: libraries/entities/src/EntityItem.cpp [hunk 11]
```diff
@@ -284,36 +321,6 @@ OctreeElement::AppendState EntityItem::appendEntityData(OctreePacketData* packet
...
-        APPEND_ENTITY_PROPERTY(PROP_CLONE_LIMIT, getCloneLimit());
-        APPEND_ENTITY_PROPERTY(PROP_CLONE_DYNAMIC, getCloneDynamic());
-        APPEND_ENTITY_PROPERTY(PROP_CLONE_AVATAR_ENTITY, getCloneAvatarEntity());
-        APPEND_ENTITY_PROPERTY(PROP_CLONE_ORIGIN_ID, getCloneOriginID());
-
-        withReadLock([&] {
-            _grabProperties.appendSubclassData(packetData, params, entityTreeElementExtraEncodeData, requestedProperties,
-                                               propertyFlags, propertiesDidntFit, propertyCount, appendState);
-        });
-
```
Comment (AndrewMeadows) [inline libraries/entities/src/EntityItem.cpp]: Note: these changes in the order by which data is packed (in `EntityItem::appendEntityData()`) must be reflected in `EntityItem::readEntityDataFromBuffer()`.  As far as I can tell: they are.  Bad Things will happen if there is order asymmetry, and odds are they will be very visible under QA scrutiny... but not necessarily very visible.

File: libraries/entities/src/EntityItem.cpp [hunk 12]
```diff
@@ -70,33 +70,60 @@ EntityItem::~EntityItem() {
...
+    requestedProperties += PROP_LOCKED;
+    requestedProperties += PROP_USER_DATA;
+    requestedProperties += PROP_HREF;
+    requestedProperties += PROP_DESCRIPTION;
     requestedProperties += PROP_POSITION;
+    requestedProperties += PROP_DIMENSIONS;
     requestedProperties += PROP_ROTATION;
-    requestedProperties += PROP_VELOCITY;
-    requestedProperties += PROP_ANGULAR_VELOCITY;
-    requestedProperties += PROP_ACCELERATION;
```
[the remark this answers is not in the document]
Comment (SamGondelman) [reply libraries/entities/src/EntityItem.cpp]: I have left these for now but yeah, we can reorder things again later.

My (perhaps too ambitious) dream is to have it so you only have to define the properties in one place one day, which will make all of that easier.  I haven't quite figured out how best to achieve that yet.

File: libraries/entities/src/EntityItem.cpp (hunk 9 above)
Comment (SamGondelman) [reply libraries/entities/src/EntityItem.cpp]: I removed this comment and added this comment in appendEntityData:
```
        // NOTE: When we enable partial packing of entity properties, we'll want to pack simulationOwner, transform, and velocity properties near each other
        // since they will commonly be transmitted together.  simulationOwner must always go first, to avoid race conditions of simulation ownership bids
```

File: libraries/entities/src/EntityItem.cpp (hunk 10 above)
Comment (SamGondelman) [reply libraries/entities/src/EntityItem.cpp]: The "are these needed" here refers specifically to whether or not we need to be doing a `COPY_ENTITY_PROPERTY_TO_PROPERTIES` here.  In general, the local variables are only used by the script layer, and are never actually sent over the wire.

File: libraries/entities/src/EntityItem.cpp (hunk 11 above)
Comment (SamGondelman) [reply libraries/entities/src/EntityItem.cpp]: :+1:

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/16222/HighFidelity-Beta-PR14585-4cd1cd6.apk

**Comment by qaAdamSmith82** [comment]
Currently Testing

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16222/HighFidelity-Beta-PR14585-4cd1cd6.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16222/HighFidelity-Beta-Interface-PR14585-4cd1cd6.exe
